parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -614,9 +614,10 @@ installKubeletKubectlAndKubeProxy() {
             if [[ -n ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then 
                 # network isolated cluster
                 echo "Detect Bootstrap profile artifact is Cache, will use oras to pull artifact binary"
-                registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/${K8S_REGISTRY_REPO}/kubernetes-node:v${KUBERNETES_VERSION}-linux-${CPU_ARCH}"
+                updateKubeBinaryRegistryURL
+                
                 K8S_DOWNLOADS_TEMP_DIR_FROM_REGISTRY="/tmp/kubernetes/downloads" # /opt folder will return permission error
-                logs_to_events "AKS.CSE.installKubeletKubectlAndKubeProxy.extractKubeBinaries" extractKubeBinaries ${KUBERNETES_VERSION} $registry_url false ${K8S_DOWNLOADS_TEMP_DIR_FROM_REGISTRY}
+                logs_to_events "AKS.CSE.installKubeletKubectlAndKubeProxy.extractKubeBinaries" extractKubeBinaries ${KUBERNETES_VERSION} "${KUBE_BINARY_REGISTRY_URL:-}" false ${K8S_DOWNLOADS_TEMP_DIR_FROM_REGISTRY}
```

Comment:
if I want to know where KUBE_BINARY_REGISTRY_URL is defined, I need to check the function of updateKubeBinaryRegistryURL.

Seems not staightforward.

Reply:
emm understand your comment. But current issue is that
1. i need one function for unit test, so func `updateKubeBinaryRegistryURL` is neccessary
2. bash function does not support return string, it needs `echo` as return. However, i need 'echo' to print log
https://github.com/Azure/AgentBaker/pull/5335#discussion_r1925631733